python/ray/tune/trial_runner.py

```diff
@@ -322,14 +323,26 @@ def __init__(self,
 
         self._callbacks = CallbackList(callbacks or [])
 
-        self._callbacks.setup()
-
         if checkpoint_period is None:
             checkpoint_period = os.getenv("TUNE_GLOBAL_CHECKPOINT_S", "auto")
 
         self._checkpoint_period = checkpoint_period
         self._checkpoint_manager = self._create_checkpoint_manager()
 
+    def setup_callbacks(self, experiment: Optional[Experiment]) -> None:
```

Comment:
Can we rename this to have a general `setup_experiments` method or something like this? I don't like the idea of explicitly calling callbacks from outside the runner, but I'm happy to run general setup and cleanup functions before and after the tuning run. In my opinion calling callbacks should be self-contained and handled only from within the  runner

Reply:
Trial runner doesn't have all the information that we want to pass to callbacks, that is why a separate method taking an experiment is required. I can rename it

Reply:
Renaming sounds fine. Can you also just pass the full experiment list to it, and the trial runner then just passes `experiments[0]`?

Reply:
Done